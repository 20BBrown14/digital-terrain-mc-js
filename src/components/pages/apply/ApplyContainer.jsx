import React from 'react';
import ApplyView from './ApplyView';

/**
 * Apply page container.
 */
const ApplyContainer = () => (
  <ApplyView />
);

export default ApplyContainer;
