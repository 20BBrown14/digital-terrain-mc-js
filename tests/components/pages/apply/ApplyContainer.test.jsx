import React from 'react';
import ApplyContainer from '../../../../src/components/pages/apply/ApplyContainer';

describe('ApplyContainer', () => {
  it('renders a default view', () => {
    const container = render(<ApplyContainer />);
    expect(container).toMatchSnapshot();
  });

  it('has an ApplyView', () => {
    const container = shallow(<ApplyContainer />);
    expect(container.find('ApplyView')).toHaveLength(1);
  });
});
