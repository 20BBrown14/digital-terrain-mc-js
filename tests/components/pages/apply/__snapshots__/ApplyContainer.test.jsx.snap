// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplyContainer renders a default view 1`] = `
<div
  class="apply-page-content"
>
  <h1
    class="apply-page-title"
  >
    Apply to Play on Digital Terrain
  </h1>
  <div
    class="apply-page-application-form"
  >
    <form
      class="ant-form ant-form-vertical"
    >
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class="ant-form-item-required"
            for="in-game-name"
            title="In-game Name"
          >
            In-game Name
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                class="ant-input"
                id="in-game-name"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class="ant-form-item-required"
            for="discord-username"
            title="Discord Username"
          >
            Discord Username
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                class="ant-input"
                id="discord-username"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class="ant-form-item-required"
            for="age"
            title="Age"
          >
            Age
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <div
                class="ant-input-number"
              >
                <div
                  class="ant-input-number-handler-wrap"
                >
                  <span
                    aria-label="Increase Value"
                    class="ant-input-number-handler ant-input-number-handler-up"
                    role="button"
                    unselectable="unselectable"
                  >
                    <span
                      aria-label="up"
                      class="anticon anticon-up ant-input-number-handler-up-inner"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        class=""
                        data-icon="up"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                  <span
                    aria-label="Decrease Value"
                    class="ant-input-number-handler ant-input-number-handler-down"
                    role="button"
                    unselectable="unselectable"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-input-number-handler-down-inner"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        class=""
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
                <div
                  class="ant-input-number-input-wrap"
                >
                  <input
                    aria-valuemax="120"
                    aria-valuemin="1"
                    autocomplete="off"
                    class="ant-input-number-input"
                    id="age"
                    max="120"
                    min="1"
                    role="spinbutton"
                    step="1"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class="ant-form-item-required"
            for="location"
            title="Location"
          >
            Location
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                class="ant-input"
                id="location"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class="ant-form-item-required"
            for="join-reasoning"
            title="Why do you want to join Digital Terrain?"
          >
            Why do you want to join Digital Terrain?
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <textarea
                class="ant-input"
                id="join-reasoning"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class="ant-form-item-required"
            for="play-style"
            title="What is your ideal play style on Digital Terrain?"
          >
            What is your ideal play style on Digital Terrain?
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <textarea
                class="ant-input"
                id="play-style"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class=""
            for="free-time"
            title="What do you like to do in your free time?"
          >
            What do you like to do in your free time?
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <textarea
                class="ant-input"
                id="free-time"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class=""
            for="source"
            title="How did you hear about Digital Terrain?"
          >
            How did you hear about Digital Terrain?
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <span
                class="ant-input-group"
              >
                <div
                  class="ant-select ant-select-single ant-select-show-arrow"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="undefined_list_0"
                        aria-autocomplete="list"
                        aria-controls="undefined_list"
                        aria-haspopup="listbox"
                        aria-owns="undefined_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        readonly=""
                        role="combobox"
                        style="opacity:0"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-placeholder"
                    >
                      Select an option
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select:none;-webkit-user-select:none"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        class=""
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
                <textarea
                  class="ant-input"
                >
                  If 'Friend' or 'Other' please provide specifics here
                </textarea>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <button
                class="ant-btn apply-page-submit-button ant-btn-primary"
                type="submit"
              >
                <span>
                  Submit
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
